src/core/src/Context.cpp

```diff
@@ -31,21 +34,88 @@
 #include <chrono>
 #endif
 
+#include <cstdlib>
+
+#if defined(__linux__)
+#include <pwd.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+#include <cerrno>
+#endif
+
 namespace cesium::omniverse {
 
+namespace {
+// Quite a lot of ceremony to get the home directory
+std::string getCacheDatabaseName() {
+    std::string homeDir;
+#if defined(__linux__)
+    if (char* cString = std::getenv("HOME")) {
+        homeDir = cString;
+    } else {
+        passwd pwd;
+        long bufsize = sysconf(_SC_GETPW_R_SIZE_MAX);
+        if (bufsize == -1) {
+            bufsize = 16384;
+        }
+        char* buf = new char[static_cast<size_t>(bufsize)];
+        passwd* result = nullptr;
+        int getResult = getpwuid_r(getuid(), &pwd, buf, static_cast<size_t>(bufsize), &result);
+        if (getResult == 0) {
+            homeDir = pwd.pw_dir;
+        }
+        delete[] buf;
+    }
+#elif defined(_WIN32)
+    if (char* cString = getenv("USERPROFILE")) {
+        homeDir = cString;
+    }
+#endif
+    if (!homeDir.empty()) {
+        std::filesystem::path homeDirPath(homeDir);
+        auto cacheFilePath = homeDirPath / ".nvidia-omniverse" / "cesium-request-cache.sqlite";
+        return cacheFilePath.generic_string();
+    }
+    return {};
+}
+
+std::shared_ptr<CesiumAsync::ICacheDatabase>& getCacheDatabase(const std::shared_ptr<Logger>& logger) {
+    // lambda is used here so that real code can be run when the static is initialized.
+    static auto pCacheDatabase = [&]() -> std::shared_ptr<CesiumAsync::ICacheDatabase> {
+        uint64_t maxCacheItems = Settings::getMaxCacheItems();
+        if (maxCacheItems == 0) {
+            logger->oneTimeWarning("maxCacheItems set to 0, so disabling accessor cache");
+            return {};
+        } else if (auto dbName = getCacheDatabaseName(); !dbName.empty()) {
+            return std::make_shared<CesiumAsync::SqliteCache>(logger, getCacheDatabaseName(), maxCacheItems);
+        }
+        logger->oneTimeWarning("could not get name for cache database");
+        return {};
+    }();
+    return pCacheDatabase;
```

Comment:
It seems like `maxCacheItems` will only get set on extension startup. Should it be possible for a user to set the value and have it update dynamically?

Reply:
That functionality -- updating the cache size after startup -- would have to be added to the cesium-native interface.

Reply:
Ah ok. Maybe the UI just needs to say `(requires restart to take effect)` or something.

Reply:
@timoore bump on the UI change (any other suggestions welcome too)